util/types/json/compare.go

```diff
@@ -88,32 +74,33 @@ func CompareJSON(j1 JSON, j2 JSON) (cmp int, err error) {
 			// for JSON null.
 			cmp = 0
 		}
-		switch x := j1.(type) {
-		case jsonLiteral:
-			left := int(x)
-			right := int(j2.(jsonLiteral))
+		switch j1.typeCode {
+		case typeCodeLiteral:
+			left := j1.i64
+			right := j2.i64
 			// false is less than true.
-			cmp = right - left
-		case jsonInt64, jsonDouble:
-			left := jsonAsFloat64(j1)
-			right := jsonAsFloat64(j2)
+			cmp = int(right - left)
```

Comment:
I also wonder compare(null, false) = ?
In most cases, compare a null value with anything else will return null.